internal/service/advancedcluster/data_source_advanced_cluster.go

```diff
@@ -313,6 +317,9 @@ func dataSourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.
 		if err := d.Set("disk_size_gb", GetDiskSizeGBFromReplicationSpec(clusterDescLatest)); err != nil {
 			return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "disk_size_gb", clusterName, err))
 		}
+		if err := d.Set("replica_set_scaling_strategy", clusterDescLatest.GetReplicaSetScalingStrategy()); err != nil {
+			return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "replica_set_scaling_strategy", clusterName, err))
+		}
```

Comment:
For the case of both data sources I believe we are only handling the case when the user defines `use_replication_spec_per_shard = true`, but if we want to handle all cases we should also consider the other path as done in the resource. Would suggest adjusting `configReplicaSetScalingStrategyOldSchema` to use data source without this option so it is captured there as well.